ffi_interface/src/lib.rs

```diff
@@ -177,6 +177,38 @@ pub fn update_commitment(
     Ok((delta_commitment + old_commitment).to_bytes_uncompressed())
 }
 
+/// Update commitment for sparse vector.
+pub fn update_commitment_sparse(
+    committer: &DefaultCommitter,
+    old_commitment_bytes: CommitmentBytes,
+    // There can only be at most 256 elements in a verkle branch
+    commitment_index_vec: Vec<usize>,
+    old_scalar_bytes_vec: Vec<ScalarBytes>,
+    new_scalar_bytes_vec: Vec<ScalarBytes>,
+) -> Result<CommitmentBytes, Error> {
+    let old_commitment = Element::from_bytes_unchecked_uncompressed(old_commitment_bytes);
+
+    let mut delta_values: Vec<(Fr, usize)> = Vec::new();
+
+    // For each index in commitment_index, we compute the delta value.
+    for index in 0..commitment_index_vec.len() {
+        let old_scalar = fr_from_le_bytes(&old_scalar_bytes_vec[index]).unwrap();
+        let new_scalar = fr_from_le_bytes(&new_scalar_bytes_vec[index]).unwrap();
+
+        let tuple = (new_scalar - old_scalar, commitment_index_vec[index]);
+
+        delta_values.push(tuple);
+    }
+
+    let delta_commitment = committer.commit_sparse(delta_values);
+
+    // If commitment is empty, then we are creating a new commitment.
+    if old_commitment_bytes == [0u8; 64] {
+        return Ok(delta_commitment.to_bytes_uncompressed());
+    }
```

Comment:
```suggestion
```

Reply:
We should instead check for the identity element instead of checking for all zeroes on the serialized commitment

Reply:
@kevaundray my reasoning was: If the trie part makes all the calculations of commitments dynamic, on the first calculation of the commitment, the commitment would be empty

Reply:
@kevaundray Do you have any idea how would we represent an identity element (point at infinity) in Java if we are using 64bytes (affine coordinates)?